blockless-cli/src/cli_clap.rs

```diff
@@ -1,68 +1,117 @@
+    "the input file is wasm file, configure file, or car file";
+
+const DEBUG_INFO_HELP: &str  = 
+    "the input file is wasm file, configure file, or car file";
+
+const APP_ARGS_HELP: &str = 
+    "the app args will pass into the app";
+
+const FS_ROOT_PATH_HELP: &str = 
+    "the root directory for the runtime.";
+
+const RUNTIME_LOGGER_HELP: &str = 
+    "the logger file for the runtime.";
+
+const LIMITED_MEMORY_HELP: &str = 
+    "the limited memory for the runtime, default is infine.";
+
+const RUN_TIME_HELP: &str = 
+    "the run time for the runtime, default is infine.";
+
+const ENTRY_HELP: &str = 
+    "the entry for wasm, default is _start";
+
+const LIMITED_FUEL_HELP: &str = 
+    "the limited fuel for runtime, default is infine";
+
+const ENVS_HELP: &str = 
+    "the app envs will pass into the app";
+
+const PERMISSION_HELP: &str = 
+    "the permissions for app";
+
+fn parse_envs(envs: &str) -> Result<(String, String)> {
+    let parts: Vec<_> = envs.splitn(2, "=").collect();
+    if parts.len() != 2 {
+        bail!("must be of the form `key=value`")
+    }
+    Ok((parts[0].to_string(), parts[1].to_string()))
 }
 
-#[rustfmt::skip]
-pub(crate) fn apply_config(cfg: &mut BlocklessConfig, matches: &ArgMatches) {
-    matches.get_one::<bool>("debug_info").map(|d| cfg.debug_info(*d));
-    matches.get_one::<String>("fs_root_path").map(|f| cfg.fs_root_path(Some(f.clone())));
-    matches.get_one::<String>("runtime_logger").map(|f| cfg.runtime_logger(Some(f.clone())));
-    matches.get_one::<String>("entry").map(|f| cfg.entry(f.clone()));
-    matches.get_one::<u64>("limited_memory").map(|f| cfg.limited_memory(Some(*f)));
-    matches.get_one::<u64>("run_time").map(|f| cfg.set_run_time(Some(*f)));
-    matches.get_one::<u64>("limited_fuel").map(|f| cfg.limited_fuel(Some(*f)));
+fn parse_permission(permsion: &str) -> Result<Permission> {
+    let url = Url::from_str(permsion)?;
+    Ok(Permission { 
+        schema: url.scheme().into(), 
+        url: permsion.into() 
+    })
+}
+
+#[derive(Parser, Debug)]
+pub(crate) struct CliCommandOpts {
+    #[clap(value_name = "INPUT", required = true, help = INPUT_HELP )]
+    input: String,
+
+    #[clap(long = "debug-info", value_name = "DEBUG-INFO", help = DEBUG_INFO_HELP)]
+    debug_info: bool,
+
+    #[clap(long = "fs-root-path", value_name = "FS-ROOT-PATH", help = FS_ROOT_PATH_HELP)]
+    fs_root_path: Option<String>,
+
+    #[clap(long = "runtime-logger", value_name = "RUNTIME-LOGGER", help = RUNTIME_LOGGER_HELP)]
+    runtime_logger: Option<String>,
+
+    #[clap(long = "limited-memory", value_name = "LIMITED-MEMORY", help = LIMITED_MEMORY_HELP)]
+    limited_memory: Option<u64>,
+
+    #[clap(long = "run-time", value_name = "RUN-TIME", help = RUN_TIME_HELP)]
+    run_time: Option<u64>,
+
+    #[clap(long = "entry", value_name = "ENTERY-HELP", help = ENTRY_HELP)]
+    entry: Option<String>,
+
+    #[clap(long = "limited-fuel", value_name = "ENTERY-HELP", help = LIMITED_FUEL_HELP)]
```

Comment:
`value_name` is `ENTERY-HELP`, should be `LIMITED-FUEL-HELP` or something like that?

Reply:
done

Reply:
thank you for you feedback.